programs/perltidy.nix

```
@@ -0,0 +1,56 @@
+{ lib, pkgs, config, ... }:
+let
+  cfg = config.programs.perltidy;
+in
+{
+  meta.maintainers = [ "haruki7049" ];
+
+  options.programs.perltidy = {
+    enable = lib.mkEnableOption "perltidy";
+    package = lib.mkPackageOption pkgs.perlPackages "PerlTidy" { };
+
+    options = lib.mkOption {
+      description = "Options for perltidy";
+      type = lib.types.listOf lib.types.str;
+      default = [
+        "-b"
+        "-bext='/'"
+      ];
+    };
+
+    perltidyrc = lib.mkOption {
+      description = "A path for perltidy's configuration file, usually named .perltidyrc";
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+    };
+
+    includes = lib.mkOption {
+      description = "Path/file patterns to include for perltidy";
+      type = lib.types.listOf lib.types.str;
+      default = [ "*.pl" ];
+    };
+    excludes = lib.mkOption {
+      description = "Path/file patterns to exclude for perltidy";
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    settings.formatter.perltidy = {
+      command = cfg.package;
+
+      options =
+        if cfg.perltidyrc == null then
+          cfg.options
+        else
+          [
+            "-pro=${cfg.perltidyrc}"
+          ] ++ cfg.options;
```

```suggestion
        lib.optional (cfg.perltidyrc != null) "-pro=${cfg.perltidyrc}"
        ++ cfg.options;
```
Something like that shows clearer intent